### PR TITLE
Fix: Allow single VLAN ranges in validation

### PR DESCRIPTION
**Pull request by gretelliz**
This PR addresses an issue in the VLAN range validation logic within `main.py`.

Previously, the assertion `assert sdx_vlan[0] < sdx_vlan[1]` prevented the use of a single VLAN defined as a range (e.g., `100:100`). This change updates the assertion to `assert sdx_vlan[0] <= sdx_vlan[1]`, allowing the start and end of the range to be equal.

**Changes**

*   Modified the validation in `main.py` to handle single VLAN ranges.
*   Increased unit tests to `tests/unit/test_main.py` to cover the following scenarios:
    *   Valid single VLAN range (e.g., `100:100`).
    *   Invalid range where the start is greater than the end (e.g., `100:99`).
    *   Invalid non-numeric and malformed range formats.

This ensures that the VLAN parsing logic is more robust and aligns with the expected behavior of supporting single VLANs specified in a range format.

**Unit tests**

Tests are passing

```
tests/unit/test_controllers.py ..                                                                                                                                                            [ 13%]
tests/unit/test_main.py .............                                                                                                                                                        [100%]

========================================================================================= warnings summary =========================================================================================
tests/unit/test_main.py: 13 warnings
  /home/student/myenv/lib/python3.11/site-packages/starlette/routing.py:885: DeprecationWarning: The `on_event` decorator is deprecated, and will be removed in version 1.0.0. Refer to https://www.starlette.io/lifespan/ for recommended approach.
    warnings.warn(

tests/unit/test_main.py: 13 warnings
  /home/student/myenv/lib/python3.11/site-packages/httpx/_client.py:1426: DeprecationWarning: The 'app' shortcut is now deprecated. Use the explicit style 'transport=ASGITransport(app=...)' instead.
    warnings.warn(message, DeprecationWarning)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
================================================================================= 15 passed, 26 warnings in 1.22s ==================================================================================
```

### CODE REVIEW

File: tests/unit/test_main.py [hunk 1]
```diff
@@ -564,6 +564,22 @@ def test_parse_vlan(self):
...
+        assert vlan is None
+        assert "Invalid vlan range" in msg
+        # case 8: range - invalid format (non-numeric)
+        vlan, msg = self.napp.parse_vlan("a:b")
+        assert vlan is None
+        assert "Invalid vlan range" in msg
+        # case 9: range - invalid format (missing colon)
+        vlan, msg = self.napp.parse_vlan("100-200")
+        assert vlan is None
+        assert "Invalid vlan range" in msg
```
Comment (viniarck) [inline tests/unit/test_main.py]: Looks like `"100.2:100.1"` would also be accepted and mapped into `[100, 100]`. Is a cast from string float to int be expected to be cast indeed or should it be a validation error too?

**Review by viniarck — COMMENTED** [review]
Nicely done @gretelliz, overall lgtm. 

I have one question about a potential corner case parsed expression.

File: tests/unit/test_main.py (hunk 1 above)
Comment (gretelliz) [reply tests/unit/test_main.py]: Hi @viniarck, thank you very much for your comments.

You are right: I added the test both for the case of strings with float and for the scenario where sdx_vlan is an int. After the changes, all tests pass.
